### DISCUSSION TITLE
Failed to get block template: Server returned rpc error Server error: RPCModuleIsDisabled

### DISCUSSION THREAD

**Discussion by pranavt61** [nervosnetwork/ckb#2425] [General]
**Environment**

- **CKB version**: docker image `nervos/ckb:v0.38.0`
- **Chain**: mainnet
- **Operating system**: Linux
- **Arch**: x64
- **Installation**: Docker-compose

**Describe the bug**

After getting the docker container running, I tried the RPC command 'get_block_template' and received an error
![Screen Shot 2020-11-25 at 7 57 41 PM](https://user-images.githubusercontent.com/22437126/100306552-8c282200-2f58-11eb-8d0d-2c1e3b0b208e.png)
{
  "jsonrpc":"2.0",
  "error":{
    "code":-4,
    "message":"RPCModuleIsDisabled: This RPC method is in the module `Miner`. Please modify `rpc.modules` and 
                           `block_assembler` in ckb.toml and restart the ckb node to enable it."
  },
  "id":2
}

This is what my ckb.toml looks like:
![Screen Shot 2020-11-25 at 7 59 46 PM](https://user-images.githubusercontent.com/22437126/100306739-ed4ff580-2f58-11eb-9aa2-eee4686e5c45.png)
![Screen Shot 2020-11-25 at 8 00 32 PM](https://user-images.githubusercontent.com/22437126/100306744-efb24f80-2f58-11eb-8cfe-8eaea03f465c.png)

Is there anything I am missing in the config?
Any help would be appreciated, thank you!

**Comment by driftluo** [comment]
Can't reproduce, please confirm that the configuration file has been modified, and the modified configuration file is used when ckb restart

**Comment by doitian** [comment]
What's your compose file? How do you modify `ckb.toml` and restart the node?

It seems that your config file is not effective, maybe:

* You have not restarted the `ckb run` process
* Your `ckb.toml` is not persisted when the process restarts.
* You have edited the wrong ckb.toml.

**Comment by pranavt61** [comment]
Here is my ckb container in the docker compose file
![Screen Shot 2020-11-30 at 4 44 12 PM](https://user-images.githubusercontent.com/22437126/100682647-64a3d180-332b-11eb-9dde-f9dd856d9fa1.png)

The ckb.toml file is in the "./ckb/" directory which is transferred into a volume. I am pretty sure the node is loading the cbk.toml because when I change a field to intentionally crash the node, it crashes:
![Screen Shot 2020-11-30 at 4 49 54 PM](https://user-images.githubusercontent.com/22437126/100683008-45f20a80-332c-11eb-9a19-b3356310c088.png)

**Comment by pranavt61** [comment]
Just found this in the logs:
`ckb-node_1    | 2020-12-01 01:21:43.303 +00:00 main INFO main  Miner is disabled because block assmebler is not a recommended lock format. Edit ckb.toml or use `ckb run --ba-advanced` to use other lock scripts`

Am I missing something in the block_assember part of the toml? Screenshot of the block_assember section in the original post

**Comment by quake** [comment]
the standard format of the `args` in block_assembler section should be 20 length of hex string, for example, '0x8d627decece439977a3a0a97815b63debaff2874'

checkout: https://docs.nervos.org/docs/basics/guides/devchain#3b-update-the-configuration

**Comment by pranavt61** [comment]
Works fine now, thanks for the help!
